oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/MappedJwtClaimSetConverter.java

```diff
@@ -0,0 +1,83 @@
+package org.springframework.security.oauth2.jwt;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.springframework.core.convert.converter.Converter;
+
+public class MappedJwtClaimSetConverter implements Converter<Map<String, Object>, Map<String, Object>> {
+	private final Map<String, Converter<JwtClaimAccessor, ?>> attributeConverters = new LinkedHashMap<>();
+
+	public MappedJwtClaimSetConverter(ConverterEntry... entries) {
+		this.attributeConverters.put(JwtClaimNames.AUD, JwtClaimAccessor::getAudience);
+		this.attributeConverters.put(JwtClaimNames.EXP, JwtClaimAccessor::getExpiresAt);
+		this.attributeConverters.put(JwtClaimNames.IAT, new IssuedAtConverter());
+		this.attributeConverters.put(JwtClaimNames.ISS, JwtClaimAccessor::getIssuer);
+		this.attributeConverters.put(JwtClaimNames.JTI, JwtClaimAccessor::getId);
+		this.attributeConverters.put(JwtClaimNames.NBF, JwtClaimAccessor::getNotBefore);
+		this.attributeConverters.put(JwtClaimNames.SUB, JwtClaimAccessor::getSubject);
+
+		for (ConverterEntry entry : entries) {
+			this.attributeConverters.put(entry.key, entry.converter);
+		}
+	}
+
+	@Override
+	public Map<String, Object> convert(Map<String, Object> source) {
+		Map<String, Object> mappedClaims = new HashMap<>(source);
+		JwtClaimAccessor claimAccessor = () -> mappedClaims;
+
+		for (Map.Entry<String, Converter<JwtClaimAccessor, ?>> entry : this.attributeConverters.entrySet()) {
+			Converter<JwtClaimAccessor, ?> converter = this.attributeConverters.get(entry.getKey());
+			if ( converter != null ) {
+				put(mappedClaims, entry.getKey(), converter.convert(claimAccessor));
+			}
+		}
+
+		return mappedClaims;
+	}
+
+	public static class ConverterEntry {
```

Comment:
I liked this because it made the constructor very simple. I also find working with var args much simpler than having to build a map and then pass it in.